### ISSUE TITLE
No trel peers any more

### ISSUE THREAD

**Issue by 3oris** [openthread/ot-br-posix#2484]
**Describe the bug** A clear and concise description of what the bug is.

Recently,
```
sudo ot-ctl trel peers list
Done
```
stopped showing trel peers. This used to work before.

Also, while
```
avahi-browse -d local _meshcop._udp --resolve | grep hostname
   hostname = [fuchsia-dce5-5bbc-xxxx.local]
   hostname = [fuchsia-ac67-8431-xxxx.local]
   hostname = [fuchsia-1c53-f9fa-xxxx.local]
   hostname = [fuchsia-d88c-7989-xxxx.local]
   hostname = [fuchsia-d8eb-46c5-xxxx.local]
   hostname = [rpi-otbr-1.local]
   hostname = [fuchsia-1c53-f9fa-xxxx.local]
   hostname = [fuchsia-d88c-7989-xxxx.local]
   hostname = [fuchsia-ac67-8431-xxxx.local]
   hostname = [fuchsia-d8eb-46c5-xxxx.local]
   hostname = [fuchsia-dce5-5bbc-xxxx.local]
   hostname = [rpi-otbr-1.local]
```
shows the OTBR in question

```
avahi-browse -d local _trel._udp --resolve |grep hostname
   hostname = [fuchsia-d88c-7989-xxxx.local]
   hostname = [fuchsia-d88c-7989-xxxx.local]
   hostname = [fuchsia-dce5-5bbc-xxxx.local]
   hostname = [fuchsia-dce5-5bbc-xxxx.local]
   hostname = [fuchsia-ac67-8431-xxxx.local]
   hostname = [fuchsia-ac67-8431-xxxx.local]
   hostname = [fuchsia-d8eb-46c5-xxxx.local]
   hostname = [fuchsia-d8eb-46c5-xxxx.local]
   hostname = [fuchsia-1c53-f9fa-xxxx.local]
   hostname = [fuchsia-1c53-f9fa-xxxx.local]
```
the OTBR in question doesn't seem to announce itself as trel peer

**To Reproduce** Information to reproduce the behavior, including:

1. Git commit id: 697cb48b32e3bceafbdd1dc97dc2f6fffc95a62a

3. IEEE 802.15.4 hardware platform: Silabs RCP: EFR32MG21

4. Build steps
```
NAT64=1 \
NAT64_SERVICE=openthread \
BACKBONE_ROUTER=1 \
BORDER_ROUTING=1 \
INFRA_IF_NAME=eth0 \
  ./script/bootstrap

OTBR_OPTIONS="-DOT_CHANNEL_MANAGER=ON \
                       -DOT_CHANNEL_MONITOR=ON \
                       -DOT_LINK_METRICS_MANAGER=ON \
                       -DOTBR_DUA_ROUTING=ON \
                       -DOTBR_RADIO_URL=spinel+hdlc+uart:///dev/ttyUSB0?uart-baudrate=460800&uart-flow-control \
                       -DOTBR_DNSSD_DISCOVERY_PROXY=ON \
                       -DOTBR_SRP_ADVERTISING_PROXY=ON" ./script/setup
```

5. Network topology: Flat network topology

**Expected behavior** A clear and concise description of what you expected to happen.

As before, OTBR should announce itself as trel peer, as a result
```
sudo ot-ctl trel peers list
```
should show the other trel peers again.

**Console/log output** If applicable, add console/log output to help explain your problem.

**Additional context** Add any other context about the problem here.

**Comment by superwhd** [comment]
Could you share the logs of your OTBR? It should be in `/var/log/syslog`.

cc @sunytt

**Comment by sunytt** [comment]
Yeah, logs would be useful. I also wonder if you need to add `-DOTBR_TREL=ON` to the OTBR_OPTIONS when building.

**Comment by 3oris** [comment]
I can get you the logs by tomorrow.

@sunytt : from

```
sudo ot-ctl trel
Enabled
Done
```

I was assuming that `-DOTBR_TREL=ON` was already derived implicitly. Also, I did not change setting from when I compiled last time in June, and it "just worked".

**Comment by 3oris** [comment]
Also, what `/etc/default/otbr-agent` log level do you need? I guess `-d6` should do the thing?

**Comment by sunytt** [comment]
Maybe we can use -d7? There are some useful debug logs for TREL. @3oris

**Comment by 3oris** [comment]
@sunytt @superwhd : here are the logs, a full start-stop cycle of the border router, letting it run for about 25s.
[thread2.log](https://github.com/user-attachments/files/16976598/thread2.log)

**Comment by superwhd** [comment]
I was trying to trace the registration of TREL DNS-SD service.

```
2024-09-12T11:03:49.961917+02:00 rpi-otbr-1 otbr-agent[4937]: 00:00:00.117 [I] TrelInterface-: Registering DNS-SD service: port:33139, txt:"xa=2ab47f9b2f91a083, xp=1261ed4221e3cf91"
```
Corresponding to here:
https://github.com/openthread/openthread/blob/2aeb8b833ba760ec29d5f340dd1ce7bcb61c5d56/src/core/radio/trel_interface.cpp#L169

After some calls it should arrive at https://github.com/openthread/ot-br-posix/blob/d5d79a87d248c10bbada0f5356b12dab59e321cd/src/trel_dnssd/trel_dnssd.cpp#L147.

However I don't see the logging line `Register %s service: port=%u, TXT=%d bytes`. 

That probably implies the `IsInitialized()` check fails: https://github.com/openthread/ot-br-posix/blob/d5d79a87d248c10bbada0f5356b12dab59e321cd/src/trel_dnssd/trel_dnssd.cpp#L152.

**Comment by 3oris** [comment]
Interesting, I see
```
2024-09-12T11:03:49.961917+02:00 rpi-otbr-1 otbr-agent[4937]: 00:00:00.117 [I] TrelInterface-: Registering DNS-SD service: port:33139, txt:"xa=2ab47f9b2f91a083, xp=1261ed4221e3cf91"
```
from
```
admin@rpi-otbr-1:~/ot-br-posix $ ag "Registering DNS-SD"
third_party/openthread/repo/src/core/radio/trel_interface.cpp
165:    LogInfo("Registering DNS-SD service: port:%u, txt:\"%s=%s, %s=%s\"", mUdpPort, kTxtRecordExtAddressKey,
```
in the logs though...

**Comment by 3oris** [comment]
@sunytt @superwhd 

OK, TREL is back! :D


So I followed your hint @superwhd , and found that `IsInitialized()` lives in
https://github.com/openthread/ot-br-posix/blob/main/src/trel_dnssd/trel_dnssd.hpp#L39
```
#if OTBR_ENABLE_TREL
```


Turns out I had TREL enabled via
```
OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE = 1
```
in my platform config.

This seemed to be enough until recently, and in fact @sunytt , it is now required to also put `-DOTBR_TREL=ON` into the `OTBR_OPTIONS`. I fiddled with `OT_TREL=ON` and `OTBR_ENABLE_TREL=ON` instead.


Thanks a lot for your help!


ps: Did a quick search on google, seems that there is no real documentation on this, btw.

**Comment by jwhui** [comment]
I'm not sure how recent a code base you were using. `OTBR_ENABLE_TREL` was added a couple years ago:
- https://github.com/openthread/ot-br-posix/pull/1109

**Comment by 3oris** [comment]
@jwhui -- thanks for pointing that out, still I don't know:

I started building OTBR in Mai and re-built quite regularly till some time in July when I went on vacation.
During that period I never had to use any of `OT_TREL=ON`, `OTBR_ENABLE_TREL=ON` or `OTBR_TREL=ON`. The only thing I did use was `OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE=1` because that is how it is documented on the Silabs site / in their SDK release PDFs, and probably also because I couldn't find any other documentation on how to enable TREL.
Recently, after I came back from holidays, I started building again, and it didn't work.

But thanks to you all that is now resolved (`OTBR_TREL=ON` has to be used for the OTBR setup script, the others don't work).

So at least I could learn something about how the build system works, how options trickle from the `scripts/setup`  script down into the code, and most importantly that from now on I would start with looking into `ot-br-posix/etc/cmake/options.cmake` and `ot-br-posix/third_party/openthread/repo/etc/cmake/options.cmake` respectively when it comes to these kind of build options.

So, if everybody is OK with it, I would close the issue. Thanks a lot, once again :)
